projects/ngx-d3timeline/src/lib/store/store.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { ActionTypes } from './actions';
+import { shareReplay, scan } from 'rxjs/operators';
+import { initialState } from './state';
+import { State } from './state';
+import { TimelineView } from '../view/timeline-view';
+import { Actions } from './actions';
+import { ScaleService } from '../scale.service';
+import { OptionsService } from '../options.service';
+
+@Injectable({ providedIn: 'root' })
+export class Store {
+  private readonly replayBuffer = 100;
+  private actionsSubject = new ReplaySubject<Actions>(this.replayBuffer);
+
+  constructor(
+    private scaleService: ScaleService,
+    private optionsService: OptionsService
+  ) {}
+
+  state$ = this.actionsSubject.pipe(
+    scan(this.reducer.bind(this), initialState),
```

Comment:
what does ```this.reducer.bind(this)``` do?

Reply:
calls the reducer with arguments of scan. Have to `bind(this)` because the reducer uses the this.